README.md

```diff
@@ -0,0 +1,47 @@
+# Gomoss
+Go client for [Moss](http://theory.stanford.edu/%7Eaiken/moss/) plagiarism software detector.
+
+## Introduction
+This is Go interface for Moss client. It is written for [omegaUp](https://omegaup.com/) project
+
+## Installation
+`go get -u github.com/SpaceWhite/gomoss`
+
+## Usage
+```
+import "github.com/SpaceWhite/gomoss"
+import "encoding/json"
+
+func main() {
+    userID := 123456789
+    request := gomoss.NewRequest(userID)
+
+    // Add code to MOSS
+    aFile, _ := os.Open("codes/a.c")
+    bFile, _ := os.Open("codes/b.c")
+    request.Code.Add(aFile, "a.c")
+    request.Code.Add(bFile, "b.c")
+
+    // Add base code to MOSS
+    sampleFile, _ := os.Open("codes/sample.c")
+    request.BaseCode.Add(sampleFile, "sample.c")
+
+    // Request to MOSS
+    url := m.Submit()
```

Comment:
what did you think of the

```go
ctx := context.Background() // I forgot to suggest this earlier, but this provides a way to cancel things.

// Send the request to Moss.
url, err := request.Submit(ctx)
if err != nil {
  panic(err)
}

// Receive the response, if ready.
results, err := gomoss.DownloadReport(ctx, url)
if err != nil {
  panic(err)
}

// View results in JSON.
for _, res := range results {
    js, _ := json.Marshal(res)
    fmt.Println(string(js))
}
```

suggestion I made earlier? that should be a lot more ergonomic and you avoid having the user care about a directory that they shouldn't care about (since that's an implementation detail and you should most likely be using temporary directories anyways).